File: apps/backend/src/main/kotlin/com/lyra/app/team/domain/TeamId.kt
package com.lyra.app.team.domain

import com.lyra.common.domain.BaseId
import java.util.*

data class TeamId(private val id: UUID) : BaseId<UUID>(id) {
    constructor(id: String) : this(UUID.fromString(id))
    companion object {
        fun create() = TeamId(UUID.randomUUID())
    }
}

File: apps/frontend/packages/domain/src/team/TeamId.ts
import { UUIDValueObject } from '@lyra/shared';

export default class TeamId extends UUIDValueObject {
  private constructor(value: string) {
    super(value);
    this.ensureIsValidUuid(value);
  }

  static random(): TeamId {
    return new TeamId(crypto.randomUUID());
  }

  public static create(value: string): TeamId {
    return new TeamId(value);
  }
}
